web_app/calendar/html_template.py

```diff
@@ -0,0 +1,178 @@
+years_route = '/calendar/years'
+
+months_route = '{}/<int:year_id>/months'.format(years_route)
+days_route = '{}/<int:month_id>/days'.format(months_route)
+hours_route = '{}/<int:day_id>/hours'.format(days_route)
+
+months_link = '{}/{{year_id}}/months'.format(years_route)
+days_link = '{}/{{month_id}}/days'.format(months_link)
+hours_link = '{}/{{day_id}}/hours'.format(days_link)
+
+year_2018_link = '{}/2018/months'.format(years_route)
+year_2019_link = '{}/2019/months'.format(years_route)
+year_2020_link = '{}/2020/months'.format(years_route)
+
+
+body_html = '''<html><head>
+                <style>
+                   table, th, td {{border: 1px solid black;border-collapse: collapse;}}
+                   a {{text-decoration: none; color: black; background-color: while; }}
+                </style>
+               </head><body>{}</body></html>'''
+
+
+# ================ HOUR ========================
+# bgcolor: #eee - busy, #ffffff - free
+
+# hour_id
+# (add) action for task creation
+hour_cell_free = '''
+<tr>
+    <th width=60 bgcolor="#ffffff">{hour_id}</th>
+    <th bgcolor="#ffffff" colspan="3"> </th>
+    <th width=60 bgcolor="#ffffff">(add)</th>
+</tr>'''
+
+# hour_id
+# task_name
+# (add) action for task creation
+hour_cell = '''
+<tr>
+        <th width=60 bgcolor="#eee">{hour_id}</th>
+        <th bgcolor="#ffffff" colspan="3">{task_name}</th>
+        <th width=60 bgcolor="#ffffff">(add)</th>
+</tr>'''
+
+# year_id (days_link)
+# month_id (days_link)
+# day_id
+# hours
+# pr_day_id, nx_day_id
+# ToDo add switch to previous and next month
+# ToDo(den) move title with previous and next (month, year, day) to out
+hour_table = '''
+    <table style="width:50%">
+        <tr>
+            <th colspan="2"><a href="{days_link}/{{pr_day_id}}/hours">{{pr_day_id}}</a></th>
+            <th><a href="{days_link}">{{day_id}}</a></th>
+            <th colspan="2"><a href="{days_link}/{{nx_day_id}}/hours">{{nx_day_id}}</a></th>
+        </tr>
+        <tr>{{hours}}</tr>
+    </table>
+'''.format(days_link=days_link)
+
+
+# ==================== DAY ===============
+# bgcolor: #eee - busy, #ffffff - free, #aaa - Day zero (from another month)
+
+# year_id (months_link)
+# month_id (months_link)
+# pr_m_id, pr_m_name - previous month
+# nx_m_id, nx_m_name - next month
+# weeks
+# ToDo(den) move title with previous and next (month, year, day) to out
+day_table = '''
+    <table style="width:50%">
+        <tr>
+            <th colspan="2"><a href="{months_link}/{{pr_m_id}}/days">({{pr_m_name}})</a></th>
+            <th colspan="3"><a href="{months_link}">{{month_name}}</a></th>
```

Comment:
Why when clicking on current month, you redirect to the page with whole year?

Reply:
This logic is implemented in standard calendar (window)